### ISSUE TITLE
Resuming training

### ISSUE THREAD

**Issue by artnoage** [jzhang38/TinyLlama#36]
I am training a 120M model from scratch because I would like to do some experiments myself. When I stop and try to resume, it requires me to drop the batch size significantly otherwise I get memory error. Any ideas why?

Also please consider making a discord server where people can discuss about the project.

**Comment by jzhang38** [comment]
Hi, Artnoage if you check the [training log](https://wandb.ai/lance777/lightning_logs/reports/metric-train_loss-23-09-04-23-38-15---Vmlldzo1MzA4MzIw?accessToken=5eu2sndit2mo6eqls8h38sklcgfwt660ek1f2czlgtqjv2c6tida47qm1oty8ik9) I actually resumed the process twice and did not notice any memory error. I am not sure why that is the case on your end. 

Thanks for your suggestions on opening a discord. I think I will open one soon.

**Comment by artnoage** [comment]
Yes I already knew that. I was just thinking maybe it had to do with the size of the network. like some missed parameter. When you did the first run, did you check the memory usage? If it is not a bit issue please leave the question open for a while, in case someone else tries it.

**Comment by jzhang38** [comment]
>When you did the first run, did you check the memory usage?

The memory usage is always 39G on my end.

**Comment by dtxwhzw** [comment]
Hi! My training crashed, and I couldn't find the code to resume training from the last saved checkpoint. How can I resume my training? How do you handle this?

**Comment by ChaosCodes** [comment]
> Hi! My training crashed, and I couldn't find the code to resume training from the last saved checkpoint. How can I resume my training? How do you handle this?

You can add `--resume your_checkpoint.pth` term in your pretraining command to resume training

**Comment by dtxwhzw** [comment]
> > Hi! My training crashed, and I couldn't find the code to resume training from the last saved checkpoint. How can I resume my training? How do you handle this?
> 
> You can add `--resume your_checkpoint.pth` term in your pretraining command to resume training

thanks, i found that

**Comment by dustinwloring1988** [comment]
@artnoage can you please post this project I would like to try to he same with 2 4060 training at home.

**Comment by artnoage** [comment]
I am not so sure what are you referring to because it is been a while.
However if you like to have a quick chat over what you want to do, you can
find me in discord with the same name (artnoage)
